app/routes/notes.tsx

```diff
@@ -6,18 +6,14 @@ import { requireUserId } from "~/session.server";
 import { useUser } from "~/utils";
 import { getNoteListItems } from "~/models/note.server";
 
-type LoaderData = {
-  noteListItems: Awaited<ReturnType<typeof getNoteListItems>>;
-};
-
 export const loader: LoaderFunction = async ({ request }) => {
```

Comment:
```suggestion
export const loader = async ({ request }) => {
```
What happens if you remove `LoaderFunction`?

Reply:
yup, need to type the args and not the full function as LoaderFunction returns `Promise<Response> | Response | Promise<AppData> | AppData`

Reply:
Though I could be convinced. I just don't remember signing up for this so I need to process it a bit...

Reply:
If I remember correctly, somebody said that we would need to choose between automatic type inference & typing the function (with possibly providing the return type)

Reply:
Ok, after looking at it more, I realize that it's required for TypeScript to infer the return type. My mistake. Fixing now.

Reply:
Even better: https://github.com/microsoft/TypeScript/issues/38511